### ISSUE TITLE
Projector: default compute PCA issue

### ISSUE THREAD

**Issue by csarron** [tensorflow/tensorboard#1204]
When clicking on the projector tab, tensorboard starts loading tensor values and computes the PCA.
The problem is that if the tensor is very large (e.g. large embeddings), default computing the PCA will take super long time, and you can do nothing but wait a few minutes at least.

I believe this is flaw for the default logic, a better option would be to let the user decide whether to compute the PCA(T-SNE, etc), for example, a button saying visualize.

 - TensorBoard version: 1.8.0
 - OS Platform and version: Ubuntu 16.04
 - Python version: 3.5

**Comment by ashudeep** [comment]
This happens even if you the tensor is already Nx2 sized!

**Comment by nfelt** [comment]
I agree that there should be some control when launching the projector about which type of analysis to do, and one of the options should be "no analysis" if the values are already 2 or 3 dimensional.  (I think the custom panel can be used to achieve this for 2 dims today, but I'm not entirely sure).  It could either be a splash screen that has you select the analysis to perform, or it could just choose one by default (like PCA today) but make it interruptible if that's not the one you want.

We may not have the bandwidth to work on this in the near future, but contributions are welcome.

**Comment by filipre** [comment]
Could you share which dimensions your tensors have? It would be interesting to know what kind of tensors are too big for Tensorboard. My current one is `2810 x 128` which already breaks the whole board.
